### PR TITLE
Check if partitions root directory remains empty after restore.

### PR DESCRIPTION
**Pull request by rodrigo-lourenco-lopes**
## Description

This PR adds a warning if data directory remains empty after restore.

The request for this warning came from a support case where the customer failed to recognize that restoring from backup was not successful. This was due to a mistake in our internal documentation which we shared with the customer. As a result, their restore wasn't effective because it didn't restore data to the Zeebe disks but to temporary disks.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #https://github.com/camunda/camunda/issues/16181

### CODE REVIEW

File: zeebe/broker/src/main/java/io/camunda/zeebe/broker/partitioning/startup/RaftPartitionFactory.java [hunk 1]
```diff
@@ -24,10 +24,12 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public final class RaftPartitionFactory {
   public static final String GROUP_NAME = "raft-partition";
-
+  private static final Logger LOG = LoggerFactory.getLogger(RaftPartitionFactory.class);
```
Comment (deepthidevaki) [inline zeebe/broker/src/main/java/io/camunda/zeebe/broker/partitioning/startup/RaftPartitionFactory.java:32]: :wrench: May be we can use `io.camunda.zeebe.broker.Loggers.SYSTEM_LOGGER`

File: zeebe/broker/src/main/java/io/camunda/zeebe/broker/partitioning/startup/RaftPartitionFactory.java [hunk 2]
```diff
@@ -41,6 +43,11 @@ public RaftPartition createRaftPartition(final PartitionMetadata partitionMetada
             .resolve("partitions")
             .resolve(partitionMetadata.id().id().toString());
     try {
+      if (FileUtil.isEmpty(partitionDirectory)) {
+        LOG.info(
+            "Root directory for partition {} is empty or does not exist, creating it if does not exist.",
```
Comment (deepthidevaki) [inline zeebe/broker/src/main/java/io/camunda/zeebe/broker/partitioning/startup/RaftPartitionFactory.java:48]: Would it be better to make the message more explicit as follows? 
```suggestion
            "Root directory '{}' for partition '{}' is empty or does not exist. The partition `{}` is starting with no pre-existing data.",
```

**Review by deepthidevaki — APPROVED** [review]
Thanks :rocket:  Just some minor comments. Please check them before merging.

File: zeebe/broker/src/main/java/io/camunda/zeebe/broker/partitioning/startup/RaftPartitionFactory.java (hunk 1 above)
Comment (rodrigo-lourenco-lopes) [reply zeebe/broker/src/main/java/io/camunda/zeebe/broker/partitioning/startup/RaftPartitionFactory.java]: Good catch 🙌

**Pull request merged**